src/Rings/solving.jl

```diff
@@ -153,37 +153,49 @@ julia> map(r->map(p->evaluate(p, r), gens(I)), rat_sols)
 ```
 """
 function rational_solutions(I::MPolyIdeal{<:MPolyRingElem})
-  gb = groebner_basis(I, ordering = lex(base_ring(I)))
   R = base_ring(I)
   K = base_ring(R)
-  if 1 in gb
-    return elem_type(base_ring(R))[]
-  end
   @req dim(I) == 0 "Dimension must be zero"
-  @assert length(gb) == ngens(base_ring(I))
-  R = base_ring(I)
-  Qx, _ = polynomial_ring(base_ring(R); cached = false)
-  rts = [elem_type(Qx)[zero(Qx) for i = gens(R)]]
-  i = ngens(R)
-  for f in gb
-    sts = Vector{elem_type(Qx)}[]
+  # note that what follows does not assume shape position of the points any more
+  G = groebner_basis(I; ordering = lex(R))
+  if 1 in G
+    return elem_type(K)[]
+  end
+  xs = collect(gens(R))
```

Comment:
Does the `collect` do anything here, other than making a redundant copy of `gens(R)` ?
```suggestion
  xs = gens(R)
```